rskj-core/src/main/java/co/rsk/peg/bitcoin/BitcoinUtils.java

```diff
@@ -82,16 +84,29 @@ public static void removeSignaturesFromTransactionWithP2shMultiSigInputs(BtcTran
         }
     }
 
-    public static Optional<Sha256Hash> findWitnessCommitment(BtcTransaction tx) {
+    public static Optional<Sha256Hash> findWitnessCommitment(BtcTransaction tx, ActivationConfig.ForBlock activations) {
         Preconditions.checkState(tx.isCoinBase());
         // If more than one witness commitment, take the last one as the valid one
         List<TransactionOutput> outputsReversed = Lists.reverse(tx.getOutputs());
 
         for (TransactionOutput output : outputsReversed) {
-            Script scriptPubKey = output.getScriptPubKey();
-            if (isWitnessCommitment(scriptPubKey)) {
-                Sha256Hash witnessCommitment = extractWitnessCommitmentHash(scriptPubKey);
-                return Optional.of(witnessCommitment);
+            try {
+                Script scriptPubKey = output.getScriptPubKey();
+                if (isWitnessCommitment(scriptPubKey)) {
+                    Sha256Hash witnessCommitment = extractWitnessCommitmentHash(scriptPubKey);
+                    return Optional.of(witnessCommitment);
+                }
+            } catch (ScriptException e) {
+                logger.debug(
+                    "[findWitnessCommitment] Failed to extract witness commitment from output {}. {}",
+                    output,
+                    e.getMessage()
+                );
+
+                if (!activations.isActive(ConsensusRule.RSKIP460)) {
+                    // Pre RSKIP460, the exception was not caught and the process could not continue
+                    throw e;
+                }
```

Comment:
Should we return an `Optional.empty()` , or will this scenario never happen?

Reply:
This is precisely what we are doing returning an `Optional.empty()` at the end.